src/main/java/entities/post.java

```diff
@@ -0,0 +1,130 @@
+package entities;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * The Post class contains all the necessary information of a post: its ID, title,
+ * description,tags,collaborators, deadline, and users who favourited it.
+ */
+public class Post extends Postable{
+    public static int numPostsCreated = 0;
+    /**The id is equal to the number of posts created.*/
+    private int id;
+    /**This is a list of the users who favourited this post.*/
+    private List<User> favouritedUsers;
+    private String title;
+    private List<String> tags;
+    private String collaborators;
+    private Date deadline;
+
+    /**
+     * @param poster the User who is posting this Post. This is an attribute of the superclass Postable.
+     * @param title this Post's title.
+     * @param mainDesc this Post's description.
+     * @param tags the tags this Post is added to.
+     * @param collaborators a description of the type of the collaborators the User who posted this Posts is looking for.
+     * @param deadline the date on which the Post is to be deleted from the database.
+     * @param replies the comments that are added to this post. This is an attribute of the superclass Postable.
+     * @param numPostsCreated1 the number of posts created so far.
+     */
+    public Post(User poster, String title, String mainDesc, List<String> tags, String collaborators,
+                Date deadline, List<Comment> replies, int numPostsCreated1){
+        super.user = poster;
+        super.body = mainDesc;
+        super.replies = replies;
+        numPostsCreated = numPostsCreated1;
```

Comment:
If numPostsCreated is static I don't think we need to assign it here or have it in the parameter list of the constructor. The increment at the end of the constructor should be fine.

Reply:
Yes, we need it in the constructor because we want to ensure data persistence so even when the program is restarted, we get numPostsCreated from the database and pass it through the constructor so we don't start at 0 again.